Tchap/Modules/Room/DataSources/RoomDataSource.m

```diff
@@ -260,54 +239,48 @@ - (void)fetchEncryptionTrustedLevel
-//        if (![event.eventId isEqualToString:self.threadId] && ![event.threadId isEqualToString:self.threadId])
-//        {
-//            //  Ignore the event
-//            return NO;
-//        }
-//        //  also ignore events related to un-threaded or events from other threads
-//        if (!event.isInThread && event.relatesTo.eventId)
-//        {
-//            MXEvent *relatedEvent = [self.mxSession.store eventWithEventId:event.relatesTo.eventId
-//                                                                    inRoom:event.roomId];
-//            if (![relatedEvent.threadId isEqualToString:self.threadId])
-//            {
-//                //  ignore the event
-//                return NO;
-//            }
-//        }
-//    }
-//    else if (RiotSettings.shared.enableThreads)
-//    {
-//        //  if not in a thread, ignore all threaded events
-//        if (event.isInThread)
-//        {
-//            //  ignore the event
-//            return NO;
-//        }
-//        //  also ignore events related to threaded events
-//        if (event.relatesTo.eventId)
-//        {
-//            MXEvent *relatedEvent = [self.mxSession.store eventWithEventId:event.relatesTo.eventId
-//                                                                    inRoom:event.roomId];
-//            if (relatedEvent.isInThread)
-//            {
-//                //  ignore the event
-//                return NO;
-//            }
-//        }
-//    }
+    if (self.threadId)
+    {
+        //  if in a thread, ignore non-root event or events from other threads
+        if (![event.eventId isEqualToString:self.threadId] && ![event.threadId isEqualToString:self.threadId])
+        {
+            //  Ignore the event
+            return NO;
+        }
+        //  also ignore events related to un-threaded or events from other threads
+        if (!event.isInThread && event.relatesTo.eventId)
+        {
+            MXEvent *relatedEvent = [self.mxSession.store eventWithEventId:event.relatesTo.eventId
+                                                                    inRoom:event.roomId];
+            if (![relatedEvent.threadId isEqualToString:self.threadId])
+            {
+                //  ignore the event
+                return NO;
+            }
+        }
+    }
+    else if (RiotSettings.shared.enableThreads)
+    {
+        //  if not in a thread, ignore all threaded events
+        if (event.isInThread)
+        {
+            //  ignore the event
+            return NO;
+        }
+        //  also ignore events related to threaded events
+        if (event.relatesTo.eventId)
+        {
+            MXEvent *relatedEvent = [self.mxSession.store eventWithEventId:event.relatesTo.eventId
+                                                                    inRoom:event.roomId];
+            if (relatedEvent.isInThread)
+            {
+                //  ignore the event
+                return NO;
+            }
+        }
+    }
```

Comment:
Why did you uncommented this code ?

Reply:
It's probably a merge issue. I'll revert this.